### PR TITLE
Python 3: lib2to3.fixes.fix_dict

### PR DESCRIPTION
**Pull request by Anthchirp**
Kicking off the work for #36 with a reasonably sized fixer.

This pull request is based on the `lib2to3.fixes.fix_dict` fixer. In Python 3 dictionary methods `.keys()`, `.values()`, etc. no longer return lists. The automatic fixer just wraps every call in `list()` to restore previous behaviour.

### CODE REVIEW

**Comment by Anthchirp** [comment]
For detailed workings see commit messages. Essentially: remove any unnecessarily added `list()` and `.keys()` again and clean up anything obviously untoward in the touched files. I left some flake8 and lgtm warnings untouched due to overlaps with pull request #38 and/or due to them being more involved.

Presumably one to review for @asmit3 @dermen @phyy-nx is this line:
https://github.com/cctbx/dxtbx/pull/40/files#diff-7d6f5eda50caf41e4480b92ec74eb1adL60
It does not serve any purpose as far as I can tell, as the return value goes nowhere and the called function does not have any (to me) obvious side effects. Judging from the context the function is also not supposed to throw an exception (otherwise it would be in the handling block just above). The function is also called in `__init__`, so the marked line appears to be genuinely superfluous.

dxtbx and dials tests all pass, so will squash-merge soon unless told not to.

**Comment by asmit3** [comment]
@Anthchirp we pushed a commit to restructure FormatXTC and subclasses in this branch. Can you please not squash that particular commit when merging ? Thanks for pointing out the issue to us.

File: datablock.py [hunk 1]
```diff
@@ -119,26 +119,18 @@ def unique_beams(self):
...
-        return list(obj.keys())
+        return list(obj)
 
     def unique_detectors(self):
-        """ Returns a list of detector objects. """
-        return list(self._unique_detectors_dict().keys())
-
-    def _unique_detectors_dict(self):
-        """ Returns an ordered dictionary of detector objects. """
+        """ Returns an ordered list of detector objects. """
```
Comment (graeme-winter) [inline datablock.py]: Method name is `unique_detectors_dict` yet returns a `list`

Though Looking at this in isolation I cannot fathom how a dictionary keyed by detectors with values of detector number was _useful_?

**Comment by Anthchirp** [comment]
@asmit Thank you for your commit. Since I want to use the Github "Squash and merge" button (which preserves the full history in this pull request, but not in the master branch) I will revert your commit here, then squash merge the pull request, then reapply your commit on top.

**Pull request merged**